controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -486,6 +488,28 @@ func (r *StorageConsumerReconciler) reconcileCephRadosNamespace(
 		} else if err := r.Delete(r.ctx, rns); client.IgnoreNotFound(err) != nil {
 			multierr.AppendInto(&combinedErr, err)
 		}
+
+		if bp.Name == util.GenerateNameForCephBlockPool(storageClusterName) &&
+			rns.Status != nil &&
+			rns.Status.Phase == rookCephv1.ConditionReady {
+			// now we are sure that rook reconciled the rns and we can remove older rns cr referring to same rns at ceph
+			rbdClaimName := fmt.Sprintf("%s-ceph-rbd", storageClusterName)
+			rbdStorageRequestName := getStorageRequestName(string(r.storageConsumer.UID), rbdClaimName)
+			rbdStorageRequestMd5Sum := md5.Sum([]byte(rbdStorageRequestName))
+
+			rnsPre419 := &rookCephv1.CephBlockPoolRadosNamespace{}
+			rnsPre419.Name = fmt.Sprintf("cephradosnamespace-%s", hex.EncodeToString(rbdStorageRequestMd5Sum[:16]))
+			rnsPre419.Namespace = r.namespace
```

Comment:
The configmap will have the radosnamespace CR name, we don't need to construct it again

Reply:
thanks!

Reply:
done.